File: math/9_PalindromeNumber/Solution.java
class Solution {
    public boolean isPalindrome(int x) {
        
        if (x < 0) {
            return false;
        }
        int dec = 1;
        while (x / dec >= 10) {
            dec *= 10;
        }
        while (x != 0) {
            int l = x / dec;
            int r = x % 10;
            if (l != r)
                return false;
            x = (x % dec) / 10;
            dec /= 100;
        }
        return true;
    }
}

File: math/9_PalindromeNumber/Solution.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        reversed = 0
        while x > reversed:
            reversed = (reversed * 10) + (x % 10)
            x = x // 10
        return True if reversed == x or reversed // 10 == x else False
